Documentation/guides/simulator.rst

```diff
@@ -0,0 +1,145 @@
+.. include:: /substitutions.rst
+.. _simulator:
+
+Simulator
+=========
+
+Apache NuttX has a simulator that can run as a regular program on Linux, Mac, and Windows computers.
+It's useful for debugging operating system features that aren't associated with particular
+device drivers— for instance the TCP/IP stack itself, a web interface or API for your
+application, or other communication protocols. It's also handy for trying out Apache NuttX without
+having a piece of embedded hardware.
+
+This guide assumes you're on Linux. It works on Windows and Mac too— if you know how,
+submit a PR the NuttX Companion to update this guide!
+
+Compiling
+---------
+
+#. Configure the Simulator
+
+   There are a lot of simulator configurations available that set you up to test various
+   operating system features.
+
+   Here we'll use the ``sim:tcpblaster`` configuration because it comes with networking
+   that is ready to use.
+
+    .. code-block:: bash
+
+       $ cd nuttx
+       $ ./tools/configure.sh sim:tcpblaster
+
+#. Compile
+
+    .. code-block:: bash
+
+       $ make clean; make
+
+Running
+-------
+
+#. Give the Simulator Privileges
+
+   On recent Linux distributions, you need to give the ``nuttx`` program the capabilities
+   (similar to permissions) to access the network:
+
+    .. code-block:: bash
+
+       $ sudo setcap cap_net_admin+ep ./nuttx
+
+#. Run the simulator:
+
+    .. code-block:: bash
+
+       $ ./nuttx
```

Comment:
I would add a nsh "help" call showing available applications and then running "hello"